src/vos/evtree.c

```diff
@@ -2422,15 +2412,17 @@ evt_common_insert(struct evt_context *tcx, struct evt_node *nd,
 }
 
 static int
-evt_common_rect_weight(struct evt_context *tcx, const struct evt_rect *rect,
-		       struct evt_weight *weight)
+evt_common_rect_weight(struct evt_context *tcx, daos_epoch_t epoch,
+		       const struct evt_rect *rect, struct evt_weight *weight)
```

Comment:
I don't quite understand this change, this function is to calculate 'weight' of a specified MBR, why do we need to pass a 'eopch' in and comparing it with the MBR epoch?

Instead of changing all these weight calculating/diff functions, I think we'd just improve the evt_select_node() to further compare the epoch distance (new rect with two MBRs) when the two MBR weight diffs are same, that looks more straightforward to me, what do you think?

Reply:
yeah, I could do that as well.   The "minor" field is kind of useless without a high epoch.    It only shows a weight difference if the new rectangle underwrites the mbr.   If we go that direction, perhaps we should just have one uint64 for "weight" and handle the epoch externally

Reply:
Ok, it doesn't work very well the way I have it.   Don't review this yet

Reply:
Ok, updated again.  I favored "disabling" minor.   Ideally, we'd add upper bound for mbr and get an actual weight difference.   But in the absence of such, the only case that really matters is overwrite (non-overwrites get sorted by the extent).   And in such cases, the distance to the start epoch of the MBR is a good heuristic for where to put the rectangle.